content/docs/setup/kubernetes/mesh-expansion/index.md

```diff
@@ -5,272 +5,280 @@ weight: 50
-cluster service names, which will be intercepted by the sidecar and forwarded.
+    {{< /text >}}
 
+5.  (Optional)  If the VM only calls services in the mesh, you can skip this step. Otherwise, add the ports the VM exposes  
+    to the cluster.env file with the following command. You can change the ports later if necessary.
+    
     {{< text bash >}}
-    $ @install/tools/setupMeshEx.sh@ generateDnsmasq
+
+    $ echo "ISTIO_INBOUND_PORTS=3306,8080" >> cluster.env
+
     {{< /text >}}
 
-    Here's an example generated file
+6.  Extract the initial keys for the service account to use on the VMs.
+
+    {{<text bash>}}
+
+    $ kubectl -n $SERVICE_NAMESPACE get secret istio.default  \
+        -o jsonpath='{.data.root-cert\.pem}' |base64 --decode > root-cert.pem
+    $ kubectl -n $SERVICE_NAMESPACE get secret istio.default  \
+        -o jsonpath='{.data.key\.pem}' |base64 --decode > key.pem
+    $ kubectl -n $SERVICE_NAMESPACE get secret istio.default  \
+          -o jsonpath='{.data.cert-chain\.pem}' |base64 --decode > cert-chain.pem
 
-    {{< text bash >}}
-    $ cat kubedns
-    server=/svc.cluster.local/10.150.0.7
-    address=/istio-mixer/10.150.0.8
-    address=/istio-pilot/10.150.0.6
-    address=/istio-citadel/10.150.0.9
-    address=/istio-mixer.istio-system/10.150.0.8
-    address=/istio-pilot.istio-system/10.150.0.6
-    address=/istio-citadel.istio-system/10.150.0.9
     {{< /text >}}
 
 ### Setting up the machines
 
-As an example, you can use the following "all inclusive" script to copy
-and install the setup:
+Next, on each VM you want to add to the mesh:
 
-{{< text bash >}}
-$ export GCP_OPTS="--zone MY_ZONE --project MY_PROJECT"
-$ export SERVICE_NAMESPACE=vm
-{{< /text >}}
+1.  Install the debian package with the Envoy sidecar:
 
-If you are running on a GCE VM, run
-
-{{< text bash >}}
-$ @install/tools/setupMeshEx.sh@ gceMachineSetup VM_NAME
-{{< /text >}}
+    {{< text bash >}}
 
-Otherwise, run
+    $ curl -L https://storage.googleapis.com/istio-release/releases/1.0.0/deb/istio-sidecar.deb > istio-sidecar.deb
+    $ dpkg -i istio-sidecar.deb
 
-{{< text bash >}}
-$ @install/tools/setupMeshEx.sh@ machineSetup VM_NAME
-{{< /text >}}
+    {{< /text >}}
 
-GCE provides better user experience since node agent can always relies on
-GCE metadata instance document to authenticate to Citadel. For everything
-else, e.g., on-prem or raw VM, we have to bootstrap a key/cert as credential,
-which typically has a limited lifetime. And when the cert expires, you have to
-rerun the above command.
+2.  Copy the cluster.env and *.pem files that you created in the previous section to the VM.
 
-Or the equivalent manual steps:
+3.  Add the IP address of the Istio gateway (which we found in the previous section) to `/etc/hosts` or to the DNS server. In our example we'll use `/etc/hosts` as it is the easiest way to get things working. The following is an example of updating an `/etc/hosts` file with the Istio gateway address:
```

Comment:
Line length and verb congruence. I would also encourage you to add a link to the step instead of the note. You can add an anchor like `{#gateway-IP}` on just above the step and then you can use the following suggestion:
```suggestion
3.  Add the IP address of the Istio gateway which we obtained in the [previous section](#gateway-IP) to the `/etc/hosts` file or to the DNS server. This example, uses the `/etc/hosts` file because it is the easiest way to get things working. The following example command updates the `/etc/hosts` file with the Istio gateway address:
```

Reply:
I'm always a bit leery of linking too much within a document, especially when you can basically see the step, but doesn't do any harm here. Done!

I'm not sure about the utility of addressing things that are obviously files (particularly etc/hosts, which if you have done any system administration at all is a very commonly used file) as "the foo file", "the bar folder", etc.